### PR TITLE
VMIController: Remove Expectations from VMI controller

### PR DESCRIPTION
**Pull request by Barakmor1**
Remove the Expectations usage from  VMI controller. It was used to prevent multiple launcher pods from being created when informers weren't synced, but this can be handled more simply with a List API call.
The Expectations code, [copied from Kubernetes](https://github.com/kubevirt/kubevirt/blob/a4aecde89accaa82d31d9bd3a0da76c763ea4308/pkg/controller/expectations.go#L18), is outdated and couple us to another project.
Removing it's use from the VMI Controller will simplify and clarify  KubeVirt codebase.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/vmi/vmi_test.go [hunk 1]
```diff
@@ -293,6 +293,12 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 		Expect(err).ToNot(HaveOccurred())
 	}
 
+	deletePod := func(pod *k8sv1.Pod) {
+		Expect(controller.podIndexer.Delete(pod)).To(Succeed())
+		err := kubeClient.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	}
```
[the remark this answers is not in the document]
Comment (dasionov) [reply pkg/virt-controller/watch/vmi/vmi_test.go:300]: Never mind After further inspection, I understand that it's necessary due to this specific part
```
	// If the pod is not found via the indexer (possibly due to sync delay), fall back to using the client
	if pod == nil {
		pod, err = controller.CurrentVMIPodFromClient(vmi, c.clientset)
		if err != nil {
			logger.Reason(err).Error("Failed to fetch pods directly from API.")
			return err
		}
	}
```

**Comment by vladikr** [comment]
Please allow me some time to review this.

**Pull request closed without merging**